codegen/golang.go

```diff
@@ -88,6 +88,13 @@ func WithUnwrapAllEmbeddedStructs() TableOptions {
 	}
 }
 
+// WithValueTypeOverride sets a function that can override the schema type for specific fields. Return `nil` to fall back to default behavior.
+func WithValueTypeOverride(resolver func(reflect.StructField) (*schema.ValueType, error)) TableOptions {
```

Comment:
nit: If we return an error now then I don't think we need a pointer anymore just `schema.ValueType` should do it.

Reply:
Yes, I had it in https://github.com/erezrokah/plugin-sdk/pull/1

Reply:
The confusion is my fault as @yevgenypats reviewed my botched rebase. The PR should be up to date now